lighthouse-core/lib/network-request.js

```diff
@@ -223,6 +242,7 @@ module.exports = class NetworkRequest {
 
     this._updateResponseReceivedTimeIfNecessary();
     this._updateTransferSizeForLightrider();
+    this._updateTimingsForLightrider();
```

Comment:
can we combine these two into one call (even if it then calls the two functions) so the control flow is clear. `this._fixLightriderWithHeaders` or whatever

Reply:
what's the benefit of combining them here? 

I actually like that it's explicitly calling out what will be touched and for what reason

Reply:
> what's the benefit of combining them here?

they're always called together, so making them rooted in a single call means it's not possible to separate them accidentally, and to keep this isolated, a clear branching point where some of the numbers from earlier are replaced by values from the headers.

I think it's totally fine to keep them as separate functions (called from that single function) with descriptive names.

Reply:
I was also thinking we could add new method called `_finalizeRequestDetails` (or `_finalizeRequestDetailsOnLoadingComplete`) which `onLoadingFailed` and `onLoadingFinished` both call.

But i agree with keeping these two LR methods separate.

Reply:
> I was also thinking we could add new method called `_finalizeRequestDetails`

heaven forfend that we keep platform-specific workarounds grouped by platform, but this would at least be an improvement